teps/0060-remote-resource-resolution.md

```diff
@@ -242,207 +251,533 @@ users can leverage those shared pipelines.
+- Validation of new `taskRef` and `pipelineRef` syntax.
+- Creation of `ResourceRequests`.
+- Behaviour of Pipelines on `ResourceRequest` success.
+- Behaviour of Tekton Pipelines on `ResourceRequest` failure.
+- Timing out remote resolution requests.
 
-Consider the following in developing a test plan for this enhancement:
-- Will there be e2e and integration tests, in addition to unit tests?
-- How will it be tested in isolation vs with other components?
+Eventually the resolution project may reach a point of maturity that
+Tekton Pipelines opts to use it for all `taskRef` and `pipelineRef`
+resolution. At this point Pipelines will need additional test coverage
+for:
 
-No need to outline all of the test cases, just the general strategy.  Anything
-that would count as tricky in the implementation and anything particularly
-challenging to test should be called out.
+- Correctly translating all `taskRefs` and `pipelineRefs` to
+  `ResourceRequests`.
+- End-to-end behaviour of all the resolvers supported by Pipelines "out
+  of the box".
 
-All code is expected to have adequate tests (eventually with coverage
-expectations).
--->
+### Tekton Resolution
+
+As part of this project being spun up a new suite of tests will be
+introduced covering the new CRD, its reconciler as well as individual
+suites for each of the resolvers.
 
 ## Design Evaluation
-<!--
-How does this proposal affect the reusability, simplicity, flexibility 
-and conformance of Tekton, as described in [design principles](https://github.com/tektoncd/community/blob/master/design-principles.md)
--->
+
+### Reusability
+
+The CRD-based approach is reusable across any Tekton project that wants
+to utilize remote resources. Submitting `ResourceRequest` objects for
+resolution is not a feature exclusively tied to Tekton Pipelines.
+Triggers, Workflows, Pipeline-as-Code and others should all be able to
+use `ResourceRequest` objects.
+
+### Simplicity
+
+1. Creating a new object kind specifically designed for handing off
+   responsibility of resolution is intended to make Tekton Pipelines'
+   job simpler. Pipelines does not need to bake-in support for git.
+   Eventually Pipelines may be able to remove all support for reading
+   Tasks and Pipelines from the cluster or Bundles entirely, relying
+   instead on resolvers.
+2. Creating new reconcilers from scratch is not trivial, however a
+   resolver does not need new code-generated libraries or even to know
+   the specifics of CRDs. A shared interface and template project can
+   alleviate the difficulty of writing new resolvers.
+
+### Flexibility
+
+- By decoupling Tekton Pipelines from the source of `Tasks` and
+  `Pipelines` it becomes possible to support any source independent of
+  the Pipelines codebase.
+- New approaches to resolution and resource handling can be built,
+  tested and productionized without burdening the Tekton Pipelines
+  project directly.
 
 ## Drawbacks
```

Comment:
Good point, I've added this as an additional drawback. I think there might be some interesting ways to help mitigate this problem but nothing springs to mind at the moment which would allow finer-grained RBAC to manage precisely which resolvers have access to specific types of `ResourceRequest`.

Reply:
I wonder if some sort of duck typing could be used here 🤔 

(e.g. the controller creates un/semi-structured requests based on a common schema, but these get resolved to individual apiVersion+kinds)